### ISSUE TITLE
Hangouts unable to text after receiving video

### ISSUE THREAD

**Issue by stacksjb** [getferdi/ferdi#368]
**Describe the bug**
Hangouts quits working after I receive a video message

My wife sent me a video, and Franz failed to load or render it, nor does it load any further messages. Furthermore, any attempts to reply simply return "Message not delivered". Logging into Hangouts via web browser shows several messages sent after the video (as well as the video), and none of the attempted messages to send.

Reloading hangouts or Franz does not fix it - the only way to resolve it is to log back out and back in.
**To Reproduce**

1. Go to Hangouts
2. While chatting with someone, have them send you a video message (a picture goes through fine)
4. It will fail to render, and you will be unable to reply and get the error "message not delivered"
<img width="261" alt="image" src="https://user-images.githubusercontent.com/2865491/74388024-3d920880-4dc8-11ea-8bd7-c838214f5d2d.png">
However, if you send messages from another computer or web browser, they may show up.

**Expected behavior**
You should be able to receive videos and reply successfully 

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. macOS, Windows 10 etc.] Mac OS X 10.15.2
- Ferdi Version [e.g. 5.4.3] 5.4.3 (5.4.3.994)

- Server Used: [e.g. https://api.getferdi.com] https://api.getferdi.com

**Additional context**
I notice multiple 401 error codes in the console for https://hangoutsusermedia-pa.clients6.google.com which likely may be related.

**Comment by pshlos46** [comment]
same issue here 

OS: Windows 10 Enterprise 1909 - 18363.752 build
Ferdi Version [e.g. 5.4.3] 5.4.4-beta.3

no server used.

It fixed itself after a couple of days, and removing and re-adding the hangouts service, but for some reason a couple of days had to pass

Probably for the conversations to pass?  And now the video is old in history and I it does not try to load it? If i scroll back up to the point of the conversation the video was sent, issue re-appears.

**Comment by BinaryShrub** [comment]
This is still happening for me as well (Mac), it comes and it goes, not sure what is going on here. I have done cmd + shift + R to clear things up and that sometimes (but usually doesnt) help. Note that I havent seen failures on hangouts.google.com which leads me to think it might be something ferdi related?

**Comment by jcolesio** [comment]
any update on this?

**Comment by bpwned** [comment]
I think this happens to me, too. I couldn't verify it but could it be that loading the preview of the video fails and then corrupts the state of the site? I tried to debug it with the Service Inspector and when I open a chat with a YT link there is a 401 error trying to access some Google/Youtube URL.

**Comment by bpwned** [comment]
I don't know how to debug this but I would bet it has something to do with security settings blocking some connections causing a corrupted state. Unfortunately I wasn't able to find out how these could be configured.

**Comment by fwenzel** [comment]
I just checked the console when this error happens and it tries to make this request:

`https://hangoutsusermedia-pa.clients6.google.com/v1/usermediaservice/videoinformation?key=...&mediaItemId.legacyPhotoId.obfuscatedUserId=...&mediaItemId.legacyPhotoId.photoId=...&alt=json`

Response is a 401:
```
code: 401
message: "Request had invalid authentication credentials. Expected OAuth 2 access token, login cookie or other valid authentication credential. See https://developers.google.com/identity/sign-in/web/devconsole-project."
status: "UNAUTHENTICATED"
```

Cross-origin security settings sound like a possible culprit indeed.

**Comment by bpwned** [comment]
What can be done about this? Unfortunately I'm not familiar with Electron at all, otherwise I'd start digging a little myself.

**Comment by fwenzel** [comment]
Interestingly the `videoinformation` URL is not requested in the Firefox version of Hangouts, but only on Chrome. In Chrome proper, it returns with a 200. Comparing request headers, they appear to be largely identical, but Ferdi is missing the `origin` header:

`origin: https://hangouts.google.com`

**Comment by bpwned** [comment]
Perhaps it would suffice to change Ferdi's user agent to Firefox?

**Comment by fwenzel** [comment]
I believe it pretends to be Firefox already but that's not sufficient. I suspect that instead of User Agent Sniffing, the app uses feature detection to do whatever it is doing there. Which is the preferred way to make such decisions but also means we cannot navigate our way around it with just user agent spoofing.

**Comment by conlon** [comment]
saw this issue yesterday on Ferdi 5.5.0.2116

**Comment by bpwned** [comment]
Is there maybe a way to disable link previews of all sorts?

**Comment by ferbar** [comment]
I just tried the development version of Ferdi with the updated electron. Hangouts is working again, however there are new bugs like links don't open.

**Comment by jcolesio** [comment]
any progress on this issue?

**Comment by SirBubbles** [comment]
Trust me, it's NOT stale. Ran into this issue just now. PLEASE do something to get around it, I won't be deleting convos every time someone sends a video link!

**Comment by fwenzel** [comment]
I suspect it  might not be a Ferdi issue so much as an Electron issue. Migrating Ferdie to a newer version of Electron might help -- but it might also involve a bunch of work that exceeds this bug's intended complexity :-/

**Comment by Crovax4444** [comment]
This is still a problem. I've switched from Franz to Ferdi and since this problem still seems to exist on both of them. On version 5.5.0 x64

**Comment by bwalia** [comment]
I am having issues with hangout and in general with Gmails apps. Message on hangout LHS is Google Hangouts is not signed in. Please refresh the page to sign in. But I tried even restarting the machine and clearing cache. I signed out using incognito mode and the problem seems to be fixed.

**Comment by Thimbolas** [comment]
Not Stale. 5/10/21 and it is still an issue. Trying to send a message in Ferdi using Hangouts and it says "Message not delivered". 
Does Ferdi not know how to fix this, and then ignore it, then come back the next month and mark it "stale"?  That is because the people at Ferdi keep ignoring it.  They have their Support packaged behind a wall at GitHub instead of having their own tech support.  And this, even after we pay for a year subscription. Very disappointed with Ferdi.

**Comment by vraravam** [comment]
@all: We are trying to unblock the nightly releases, but in the meantime, some of the issues reported above might have been fixed already in the development branch. For macos users, could I request you to please download the latest applications from [this folder](https://drive.google.com/drive/folders/1mJmKbhpVHdb066IHlrdYiD8TG9wlmuBv) - and report if it fixes the issues.
Also, specific to services/recipes, there is some new [self-help documentation](https://github.com/vraravam/ferdi/tree/docs-recipes-self-help#troubleshooting-recipes-self-help) that might also be needed to fix the issue(s).

As I have requested, once you try these, please report back on what issue is still outstanding. Thank you for your patience.

**Comment by vraravam** [comment]
Sorry - this PR has now been merged into the main Ferdi repo: https://github.com/getferdi/ferdi/#troubleshooting-recipes-self-help

**Comment by kris7t** [comment]
>  They have their Support packaged behind a wall at GitHub instead of having their own tech support. And this, even after we pay for a year subscription. Very disappointed with Ferdi.

I think you're mixing up Ferdi with [Franz](https://github.com/meetfranz/franz), which does have a paid subscription. In fact, Ferdi was created as a hard-fork of Franz to fix some if its issues, including the need for a paid subscription and giving your data to third parties. However, there are only a few (volunteer and entirely unpaid) people behind Ferdi for now, so fixing the issues may take a long time. Especially for stuff like Google services, where Google is [actively working to block "unofficial" browsers](https://groups.google.com/a/chromium.org/forum/#!msg/embedder-dev/STyM5ZNTHMM/POj1v_cqBgAJ).
